allennlp/modules/token_embedders/bidirectional_language_model_token_embedder.py

```diff
@@ -0,0 +1,170 @@
+
+        overrides = {
+                "model": {
+                        "contextualizer": {
+                                "return_all_layers": True
+                        }
+                }
+        }
+
+        # Load LM and the associated config.
+        archive = load_archive(archive_file, overrides=json.dumps(overrides))
+        self._lm: BidirectionalLanguageModel = archive.model
+        # pylint: disable=protected-access
+        self._lm._softmax_loss = None
+        config = archive.config
+        dict_config = config.as_dict(quiet=True)
+
+        # Extract the name of the tokens that the LM was trained on.
+        token_names = list(dict_config["model"]["text_field_embedder"]["token_embedders"].keys())
+        if len(token_names) != 1:
+            # A TokenEmbedder embeds a single set of indices. If the LM was trained with multiple embedders, and
+            # thus indices, we won't be able to embed a single set of indices here.
+            #
+            # Note: We only care about embedded indices. This does not include "tokens" which is just used to
+            # compute the loss in BidirectionalLanguageModel.
+            raise RuntimeError(f"LM from {archive_file} trained with multiple embedders!")
+        self._token_name = token_names[0]
+
+        # TODO(brendanr): Remove this hack once we have proven we can train without using the
+        # ELMoTokenCharactersIndexer. For the equivalent hack in the ELMo embedder see
+        # https://github.com/allenai/allennlp/blob/master/allennlp/modules/elmo.py#L590.
+        if bos_eos_tokens:
+            dataset_reader_config = config.get("dataset_reader")
+            if dataset_reader_config.get("type") == "multiprocess":
+                dataset_reader_config = dataset_reader_config.get("base_reader")
+            token_indexer_config = dataset_reader_config.get("token_indexers").get(self._token_name)
+            token_indexer: TokenIndexer = TokenIndexer.from_params(token_indexer_config)
+            token_list = [Token(token) for token in bos_eos_tokens]
+            bos_eos_indices = token_indexer.tokens_to_indices(token_list, self._lm.vocab, "key")["key"]
+            self._bos_indices = torch.Tensor(bos_eos_indices[0])
+            self._eos_indices = torch.Tensor(bos_eos_indices[1])
+        else:
+            self._bos_indices = None
+            self._eos_indices = None
+
+        if dropout:
+            self._dropout = torch.nn.Dropout(dropout)
+        else:
+            self._dropout = lambda x: x
+
+        self._remove_bos_eos = remove_bos_eos
+        num_layers = self._lm._contextualizer.num_layers + 1
+        self._scalar_mix = ScalarMix(mixture_size=num_layers, do_layer_norm=False, trainable=True)
+
+        character_dim = self._lm._text_field_embedder.get_output_dim()
+        contextual_dim = self._lm._contextualizer.get_output_dim()
+
+        if not contextual_dim % character_dim == 0 or character_dim > contextual_dim:
```

Comment:
couldn't this check just be `if contextual_dim % character_dim != 0`, since if character_dim is greater than contextual_dim, then the mod won't be zero (unless contextual_dim is 0, but that can't happen, can it?)